src/Common/Config/ConfigProcessor.cpp

```diff
@@ -792,6 +812,21 @@ void ConfigProcessor::decryptEncryptedElements(LoadedConfig & loaded_config)
 
 #endif
 
+XMLDocumentPtr ConfigProcessor::hideElements(LoadedConfig & loaded_config)
+{
+    XMLDocumentPtr new_preprocessed_xml = new Poco::XML::Document;
+
+    for (Node * node = loaded_config.preprocessed_xml->firstChild(); node; node = node->nextSibling())
```

Comment:
So l. 817-825 copy of the preprocessed XML structucture, right? Why is that necessary, can't we call `hideRecursive()` directly on the existing preprocessed XML?

Reply:
In the first commits I applied `hideRecursive()` directly on loaded_config.preprocessed_xml. But then I discovered that loaded_config.configuration in fact is a pointer to `loaded_config.preprocessed_xml`. So cutting nodes from `preprocessed_xml` also influences on `configuration` (missing values will be reset to defaults). That why I:
1. create copy of  `preprocessed_xml` 
2. hide nodes on copy.
3. save copy as preprocessed config file.

Reply:
Okay, but please add a short comment to the code then.

Reply:
Done. Added comment. Again. Please text a look at the comment's text and it's position. Maybe we may improve text or put it into more clear position.